### PR TITLE
Remove TFM parsing from xunit test target

### PR DESCRIPTION
**Pull request by natemcmaster**
Requires update to projects that use .NET Framework in test projects. These should be updated to only include .NET Framework when building/testing on Windows.

### CODE REVIEW

**Comment by natemcmaster** [comment]
I made the changes affected repos to ensure net451 tests are only compiled on Windows. Will push once this is merged

**Comment by natemcmaster** [comment]
:x: blocked. Conditional `TargetFrameworks` properties cause VS to explode

image

**Comment by natemcmaster** [comment]
When this https://github.com/dotnet/roslyn-project-system/pull/1444 makes it into a build, we can return to conditioning TargetFrameworks on OS.

**Comment by natemcmaster** [comment]
🔔 Updated and rebased. Ready for review.

File: build/targets/standard-lifecycle.targets [hunk 1]
```diff
@@ -125,7 +159,51 @@ that matches "$(RepositoryRoot)/shared/*.Sources".
...
+    <!-- TODO if VS Test doesn't make minimal output the default log setting, we can set that here. cref https://github.com/Microsoft/vstest/issues/301 -->
+    <VSTestLogger Condition=" '$(VSTestLogger)'=='' AND '$(TEAMCITY_VERSION)' != '' ">trx</VSTestLogger>
+    <IgnoreFailingTestProjects>false</IgnoreFailingTestProjects>
+    <ContinueOnTestError Condition="'$(KOREBUILD_IGNORE_DOTNET_TEST_EXIT_CODE)' == '1'">true</ContinueOnTestError>
+  </PropertyGroup>
+
+  <Target Name="TestProjects">
+
+    <ItemGroup Condition="'@(ProjectsToTest)'==''">
+      <!-- put unit test projects ahead of functional tests -->
```
Comment (pranavkm) [inline build/targets/standard-lifecycle.targets]: Do we actually have this as a convention? `Tests` -> Functional? Otherwise this is kinda misleading.

Comment (pranavkm) [reply build/targets/standard-lifecycle.targets]: Plus we don't super care about ordering. Ideally we should try and these in parallel.

File: build/targets/standard-lifecycle.targets [hunk 2]
```diff
@@ -125,7 +159,51 @@ that matches "$(RepositoryRoot)/shared/*.Sources".
...
+      <ProjectsToTest Include="$(RepositoryRoot)test\*\*.Tests.csproj" Exclude="$(ExcludeFromTest)" />
+      <ProjectsToTest Include="$(RepositoryRoot)test\*\*.Test.csproj" Exclude="$(ExcludeFromTest)" />
+      <ProjectsToTest Include="$(RepositoryRoot)test\*\*.csproj" Exclude="@(ProjectsToTest);$(ExcludeFromTest)" />
```
Comment (pranavkm) [inline build/targets/standard-lifecycle.targets]: Wouldn't this suffice? Also, why not make `ExcludeFromTest` an item group?

File: build/targets/standard-lifecycle.targets [hunk 3]
```diff
@@ -125,7 +159,51 @@ that matches "$(RepositoryRoot)/shared/*.Sources".
...
+
+Runs the VSTest on all projects in the ProjectsToTest itemgroup.
+###################################################################
+-->
+
+  <PropertyGroup>
+    <!-- TODO if VS Test doesn't make minimal output the default log setting, we can set that here. cref https://github.com/Microsoft/vstest/issues/301 -->
+    <VSTestLogger Condition=" '$(VSTestLogger)'=='' AND '$(TEAMCITY_VERSION)' != '' ">trx</VSTestLogger>
+    <IgnoreFailingTestProjects>false</IgnoreFailingTestProjects>
+    <ContinueOnTestError Condition="'$(KOREBUILD_IGNORE_DOTNET_TEST_EXIT_CODE)' == '1'">true</ContinueOnTestError>
```
Comment (pranavkm) [inline build/targets/standard-lifecycle.targets]: Should this be `IgnoreFailingTestProjects`? `ContinueOnTestError` isn't used elsewhere.

**Review by pranavkm — APPROVED** [review]

File: build/targets/standard-lifecycle.targets (hunk 3 above)
Comment (natemcmaster) [reply build/targets/standard-lifecycle.targets]: Yup, good catch. Missed that rename.

File: build/targets/standard-lifecycle.targets (hunk 1 above)
Comment (natemcmaster) [reply build/targets/standard-lifecycle.targets]: This is something EF has requested. Running unit tests first to catch errors early is nice, esp when functional tests can take a long time to run.

File: build/targets/standard-lifecycle.targets (hunk 2 above)
Comment (natemcmaster) [reply build/targets/standard-lifecycle.targets]: And item group might be better. I ran into this https://github.com/Microsoft/msbuild/issues/1576 which means setting ExcludeFromTest in repo.targets correctly is not easy.

Comment (natemcmaster) [reply build/targets/standard-lifecycle.targets]: Hmm, nvm. ItemGroups and remove syntax suffer from the same issue.

File: build/targets/standard-lifecycle.targets (hunk 1 above)
Comment (pranavkm) [reply build/targets/standard-lifecycle.targets]: Can we explicit about it - requiring functional tests to include the term `FunctionalTests`? Alternatively have `repo.props` determine order if it's necessary

File: build/targets/DefaultItems.targets [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    <Solutions Remove="@(ExcludeSolutions)" />
+
+    <ProjectsToPack Include="$(RepositoryRoot)src\*\*.csproj" />
+    <ProjectsToPack Remove="@(ExcludeFromPack)" />
+
+    <SharedSourceDirectories Include="$([System.IO.Directory]::GetDirectories(&quot;$(SharedSourcesFolder)&quot;, '*.Sources'))" Condition="Exists('$(SharedSourcesFolder)')" />
+    <SharedSourceDirectories Remove="@(ExcludeSharedSourceDirectories)" />
+
+    <!-- put unit test projects ahead of functional tests -->
+    <ProjectsToTest Include="$(RepositoryRoot)test\*\*.Tests.csproj" />
```
Comment (pranavkm) [inline build/targets/DefaultItems.targets]:
```xml
<ItemGroup Condition="'@(ProjectsToTest)'==''">
  <_FunctionalTests Include="$(RepositoryRoot)test\*FunctionalTest*.csproj" />
  <ProjectsToTest Include="$(RepositoryRoot)test\*Test*.csproj" Exclude="@(_FuctionalTests)" />
  <ProjectsToTest Include="@(_FunctionalTests)"  />
 ...
</ItemGroup>
```

Comment (natemcmaster) [reply build/targets/DefaultItems.targets]: 👍 

Although, the thinking behind not doing `Condition="'@(ProjectsToTest)'==''"` is that the most common scenario here is that we will want to exclude a project from testing `ExcludeFromTest` that was picked up by default globs. Redefining the default globs is cumbersome.

Comment (pranavkm) [reply build/targets/DefaultItems.targets]: Sure, we could add it if we need it later on.

**Pull request merged**